tests/test_optimize.py

```diff
@@ -462,6 +468,113 @@ def test_adam_optimizer_multivar(self, bunch, tol):
                 x_twosteps_target = x_onestep - adapted_stepsize * firstmoment / (np.sqrt(secondmoment) + 1e-8)
                 assert x_twosteps == pytest.approx(x_twosteps_target, abs=tol)
 
+    @pytest.mark.parametrize('x_start', x_vals)
+    def test_rotosolve_optimizer_univar(self, x_start, bunch, tol):
+        """Tests that rotosolve optimizer takes one and two steps correctly
+        for uni-variate functions."""
+
+        for f in univariate_funcs:
+            x_onestep = bunch.rotosolve_opt.step(f, x_start)
+            x_onestep_target = self.rotosolve_step(f, x_start)
+
+            assert x_onestep == pytest.approx(x_onestep_target, abs=tol)
+
+            x_twosteps = bunch.rotosolve_opt.step(f, x_onestep)
+            x_twosteps_target = self.rotosolve_step(f, x_onestep_target)
+
+            assert x_twosteps == pytest.approx(x_twosteps_target, abs=tol)
+
+    def test_rotosolve_optimizer_multivar(self, bunch, tol):
+        """Tests that rotosolve optimizer takes one and two steps correctly
+        for multi-variate functions."""
+
+        for f in multivariate_funcs:
+            for jdx in range(len(x_vals[:-1])):
```

Comment:
Minor suggestion (not too crucial): find this one easier to parse personally:
```suggestion
            for jdx in range(len(x_vals)-1):
```

Reply:
I agree, although all the other tests use ``len(x_vals[:-1])`` in a similar fashion. In that case it would be preferable to change all instances of this in the test file.

Reply:
Got ya, that's a good point! Well, as this was just a minor suggestion, feel free to leave it as is :)